### PR TITLE
Model initializing and utilities for generating batch files

### LINKED ISSUE #1: utility for generating batch files of the GEFS deterministic mermber
We need to train the GraphCast model for the GEFS. therefore we need to generate batch files from GEFS reanalysis deterministc member. The script would be very similar to the one for gdas-graphcast model and it should be located in the `training` folder under MLGEFS repository.
https://github.com/NOAA-EMC/graphcast/blob/main/NCEP/gdas_utility.py

### PR DESCRIPTION
**Pull request by sadeghitabas**
## Description
This PR provides scripts for preprocessing (extracting, processing grib2 files, downsampling from 0.25-degree to 1.0-degree, making netcdf files) and generating batches that are required for MLGEFS model training. It also initiates the GraphCast model as the core of the MLGEFS which will be utilized through an ensemble-based cascaded modeling procedure. The generated batches will have 1.0-degree spatial and 6-hourly temporal resolutions (4 cycles per day). Every batch file will contain multiple atmospheric variables across 13 pressure levels as well as several surface variables for one cycle. 

### Associated GraphCast Issue to close
- Closes https://github.com/NOAA-EMC/MLGEFS/issues/1

### Input data
- [ ] No changes are expected to input data.
- [x] Changes are expected to input data:
  - [x] New input data.
  - [ ] Updated input data.

### Needed libraries

- xarray
- numpy
- wgrib2
-

### CODE REVIEW

**Comment by sadeghitabas** [comment]
@junwang-noaa Please add a new project for MLGEFS in the model infrastructure.

File: training/downsampler.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# Sadegh Tabas, initial commit
+
```
Comment (junwang-noaa) [inline training/downsampler.sh:4]: @SadeghTabas-NOAA Can you add a short description?

File: training/generate_batch_files.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
+'''
+Description
```
Comment (junwang-noaa) [inline training/generate_batch_files.py:2]: Here too

File: training/generate_batch_files.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            'latitude': 'lat',
+            'longitude': 'lon',
+            'plevel': 'level',
+            'HGT_surface': 'geopotential_at_surface',
+            'LAND_surface': 'land_sea_mask',
+            'PRMSL_meansealevel': 'mean_sea_level_pressure',
+            'TMP_2maboveground': '2m_temperature',
+            'UGRD_10maboveground': '10m_u_component_of_wind',
+            'VGRD_10maboveground': '10m_v_component_of_wind',
+            'APCP_surface': 'total_precipitation_6hr',
```
Comment (junwang-noaa) [inline training/generate_batch_files.py]: Is it OK we don't have APCP?

File: training/downsampler.sh (hunk 1 above)
Comment (sadeghitabas) [reply training/downsampler.sh:4]: Done!

File: training/generate_batch_files.py (hunk 2 above)
Comment (sadeghitabas) [reply training/generate_batch_files.py:2]: Done!

File: training/generate_batch_files.py (hunk 3 above)
Comment (sadeghitabas) [reply training/generate_batch_files.py]: Updated!

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**